supporting/concentration_layer.py

```diff
@@ -231,6 +241,77 @@ def process_emissions(self, emis, isrm_obj, verbose, output_dir, output_emis_fla
             
         return tmp_dct['PM25'], tmp_dct['NH3'], tmp_dct['VOC'], tmp_dct['NOX'], tmp_dct['SOX']
     
+    def visualize_emissions(self, pollutant_name=''):
+
+        ''' Create a 5-panel plot of total emissions for each pollutant and save as a PNG file '''
+        
+        if self.verbose:
+            logging.info('- Drawing map of total emissions by pollutant.')
+
+        # Read in CA boundary
+        ca_shp = gpd.read_feather(self.shp_path)
+        ca_prj = ca_shp.to_crs(self.crs)
+        
+        # Reproject output_region
+        output_region = self.output_region.to_crs(self.crs)
+
+        pollutants = {
+            'Primary PM2.5': self.PM25e,
+            'NH3': self.NH3e,
+            'NOx': self.NOXe,
+            'SOx': self.SOXe,
+            'VOC': self.VOCe
+        }
+
+        fig, axes = plt.subplots(nrows=1, ncols=5, figsize=(22,6))
+        
+        for ax, (pol, data) in zip(axes, pollutants.items()):
+            
+            # Clip to output region if provided
+            if output_region is not None:
+                data = gpd.clip(data, output_region)
```

Comment:
Is there any way to do this clip outside of the for-loop? The function is taking a long time, and I'm trying to identify a few ways to speed it up